plugins/woocommerce-blocks/packages/components/checkbox-list/checkbox-list-option-control.tsx

```diff
@@ -0,0 +1,61 @@
+/**
+ * External dependencies
+ */
+import { useInstanceId } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import { CheckboxControl } from '../checkbox-control';
+import type { CheckboxListOptions } from './types';
+
+export type CheckboxListOptionControlProps = {
```

Comment:
Is there any benefit on exporting this type instead of keeping it internal to this file?

Reply:
Hey, @Aljullu thanks for your review. Good question!

About this, I always export props and components from its own file, in fact I always prefer to have a file to keep the types and another to extract the components, that means that I always need  to export anything to be used across files. But that does not means it is "fully" public, that is the reason why I also export what I consider fully public in the index file at the root of the folder containing the component and its types I really want to expose, I consider the rest, private or "partially" public since the access is only within the component relative folder. 
For this and other reasons like reusing types, components and styles. This has been always the structure I use for any react component I had wrote.
```
+component-name
    - ComponentName.tsx
    - types.ts
    - styles.scss
    - index.ts
```
With this structure component files are way more short and readable, since any metadata like types or styles are outside of the component main file.
And the index file is used only to determine what are the artifacts that are explicitly exposed outside.

I know this is not the approach we are using here in general but I think it should change at least a little, since types are widely used specially if we need to extend a component functionality and we should not be forced to import the whole component file just to get its type (or typeof it), or create a brand new types file to expose the type we need at that moment we need it because that means touching zones beyond the scope of the issue we are currently working on.

Well these are some of my reasons why I picked exporting this type. But after this, if you still think we should not do that and prefer the blacklist over the whitelist solution, let me know and I will change this for sure.

Reply:
Makes sense. Yeah, no need to change anything, your explanation makes sense. :+1: